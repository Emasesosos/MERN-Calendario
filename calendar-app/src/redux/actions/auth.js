import Swal from "sweetalert2";
import { fetchSinToken, fetchConToken } from "../../helpers/fetch"; // fetchCorreo,
import { types } from "../types/types";

/* ***** Acciones de Auth: Usuario ***** */

// Inicio Login Usuario
export const startLogin = (email, password) => {

    return async(dispatch) => { // Thunk

        // console.log(email, password);
        const resp = await fetchSinToken('auth', { email, password }, 'POST');
        const body = await resp.json();
        // console.log(body);

        if (body.ok) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        } else {
            Swal.fire('Error', body.msg, 'error');
        }

    };

};

// Start Register
export const startRegister = (name, email, password) => {

    return async(dispatch) => { // Thunk

        // console.log(name, email, password);
        const resp = await fetchSinToken('auth/new', { name, email, password }, 'POST');
        const body = await resp.json();
        // console.log(body);

        if (body.ok) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            // const sendCorreo = await fetchCorreo('auth/send-mail', { email }, 'POST');
            // const bodyCorreo = await sendCorreo.json();
            // console.log(bodyCorreo);
            // Swal.fire({
            //     icon: 'success',
            //     title: 'Se creÃ³ nuevo Usuario',
            //     text: bodyCorreo.msg
            // });
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        } else {
            Swal.fire('Error', body.msg, 'error');
        }

    };

};

// Start Restablish Password
export const startRestablishPass = (email, password) => {

    return async(dispatch) => { // Thunk

        // console.log(email, password);
        const resp = await fetchSinToken('auth/restablish-pass', { email, password }, 'PUT');
        const body = await resp.json();
        // console.log(body);

        if (body.ok) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            // const sendCorreo = await fetchCorreo('auth/resend-mail', { email }, 'POST');
            // const bodyCorreo = await sendCorreo.json();
            // // console.log(bodyCorreo);
            // Swal.fire({
            //     icon: 'success',
            //     title: body.msg,
            //     text: bodyCorreo.msg
            // });
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        } else {
            Swal.fire('Error', body.msg, 'error');
        }

    };

};

// Start Checking
export const startChecking = () => {

    return async(dispatch) => {

        const resp = await fetchConToken('auth/renew');
        const body = await resp.json();
        // console.log(body);

        if (body.ok) {
            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        } else {
            // Swal.fire('Error', body.msg, 'error');
            dispatch(checkingFinish());
        }

    };

};

const checkingFinish = () => {

    return {
        type: types.authCheckingFinish
    };

};

// Login Usuario
const login = (user) => {

    return {
        type: types.authLogin,
        payload: user
    };

};

// Start Logout
export const startLogout = () => {

    return (dispatch) => {
        localStorage.clear();
        dispatch(logout());
    };

};

// Logout
const logout = () => {

    return {
        type: types.authLogout,
    };

};