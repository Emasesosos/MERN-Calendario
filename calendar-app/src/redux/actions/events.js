import { types } from "../types/types";

/* ***** Acciones de CELENDAR: Events ***** */

// Agregar nuevo Evento
export const eventAddNew = (event) => {
    return {
        type: types.eventAddNew,
        payload: event
    };
};

// Activar Evento
export const eventSetActive = (event) => {
    return {
        type: types.eventSetActive,
        payload: event
    };
};

// Limpiar Nota Activa
export const eventClearActiveEvent = () => {
    return {
        type: types.eventClearActiveEvent,
    };
};

// Actualizar Nota
export const eventUpdated = (event) => {
    return {
        type: types.eventUpdated,
        payload: event
    }
};

// Eliminar Nota
export const eventDeleted = () => {
    return {
        type: types.eventDeleted,
    }

};