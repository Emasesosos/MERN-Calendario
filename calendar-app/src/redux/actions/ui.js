import { types } from "../types/types";

/* ***** Acciones de UI ***** */
// Abrir Modal
export const uiOpenModal = () => {
    return {
        type: types.uiOpenModal,
    }
};
// Cerrar Modal
export const uiCloseModal = () => {
    return {
        type: types.uiCloseModal,
    }
};

/* ***** Acciones de AUTH ***** */
/* ***** Acciones de CELENDAR ***** */
