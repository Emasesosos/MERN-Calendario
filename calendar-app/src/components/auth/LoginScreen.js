import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import useForm from '../../hooks/useForm';
import { startLogin } from '../../redux/actions/auth';
import './login.css';

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const initialForm = {
        email: 'emasesosos@gmail.com',
        password: '123456',
    };
    
    const [ formValues, handleInputChange ] = useForm(initialForm);

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        // console.log(formValues);
        dispatch(startLogin(email, password));
    };

    return (
        <div className="container login-container">
            <div className="row">
                <div className="col-md-6 login-form-1">
                    <h3>Ingreso</h3>
                    <form
                        onSubmit={ handleLogin }
                    >
                        <div className="form-group">
                            <input 
                                type="text"
                                className="form-control"
                                placeholder="Correo"
                                name="email"
                                value={ email }
                                onChange={ handleInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contraseña"
                                name="password"
                                value={ password }
                                onChange={ handleInputChange }
                            />
                        </div>
                        <div className="form-group btn-auth">
                            <input 
                                type="submit"
                                className="btnSubmit"
                                value="Login" 
                            />
                        </div>

                        <Link 
                            // to="/auth/register"
                            to="/register"
                            className="btnSubmit btn-auth"
                        >
                            Crear Nueva Cuenta
                        </Link>

                        <Link 
                            // to="/auth/register"
                            to="/restablish"
                            className="btnSubmit btn-auth mt-3"
                        >
                            ¿Olvidaste tu Contraseña?
                        </Link>

                    </form>
                </div>
            </div>
        </div>
    );
};