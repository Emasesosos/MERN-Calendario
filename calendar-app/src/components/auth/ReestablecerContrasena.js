import React from 'react';
import { useDispatch } from 'react-redux';
import Swal from 'sweetalert2';
import useForm from '../../hooks/useForm';
import { startRestablishPass } from '../../redux/actions/auth';
// import './login.css';

export const ReestablecerContrasena = () => {

    const dispatch = useDispatch();

    const initialForm = {
        email: 'correo@correo.com',
        password1: '123456',
        password2: '123456',
    };
    
    const [ formValues, handleInputChange ] = useForm(initialForm);

    const { email, password1, password2 } = formValues;

    const handleRestablishPass = (e) => {
        e.preventDefault();

        if(password1.length < 6) {
            return Swal.fire('Error', 'La contraseña debe de ser de al menos 6 caracteres', 'error');
        }

        if(password1 !== password2) {
            return Swal.fire('Error', 'Las contraseñas deben de ser iguales', 'error');
        }
        // console.log(formValues);
        dispatch(startRestablishPass(email, password1));
    };

    return (
        <div className="container login-container">
            <div className="row">
                <div className="col-md-6 login-form-2">
                    <h3>Restablecer Contraseña</h3>
                    <form
                        onSubmit={ handleRestablishPass }
                    >
                        <div className="form-group">
                            <input
                                type="email"
                                className="form-control"
                                placeholder="Correo"
                                name="email"
                                value={ email }
                                onChange={ handleInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Nueva Contraseña" 
                                name="password1"
                                value={ password1 }
                                onChange={ handleInputChange }
                            />
                        </div>

                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Repita la Contraseña" 
                                name="password2"
                                value={ password2 }
                                onChange={ handleInputChange }
                            />
                        </div>

                        <div className="form-group btn-auth">
                            <input 
                                type="submit" 
                                className="btnSubmit" 
                                value="Cambiar Contraseña" />
                        </div>

                    </form>
                </div>
            </div>
        </div>
    );
};
