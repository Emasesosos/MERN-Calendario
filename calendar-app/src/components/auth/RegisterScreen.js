import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import Swal from 'sweetalert2';
import useForm from '../../hooks/useForm';
import { startRegister } from '../../redux/actions/auth';

export const RegisterScreen = () => {

    const dispatch = useDispatch();

    const initialForm = {
        name: 'Emasesosos',
        email: 'correo@correo.com',
        password1: '123456',
        password2: '123456',
    };
    
    const [ formValues, handleInputChange ] = useForm(initialForm);

    const { name, email, password1, password2 } = formValues;

    const handleRegister = (e) => {
        e.preventDefault();
        // console.log(formValues);
        if(password1.length < 6) {
            return Swal.fire('Error', 'La contraseña debe de ser de al menos 6 caracteres', 'error');
        }

        if(password1 !== password2) {
            return Swal.fire('Error', 'Las contraseñas deben de ser iguales', 'error');
        }
        // console.log(formValues);
        dispatch(startRegister(name, email, password1));
    };

    return (
        <div className="container login-container">
            <div className="row">
                <div className="col-md-6 login-form-2">
                    <h3>Registro</h3>
                    <form
                        onSubmit={ handleRegister }
                    >
                        <div className="form-group">
                            <input
                                type="text"
                                className="form-control"
                                placeholder="Nombre"
                                name="name"
                                value={ name }
                                onChange={ handleInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="email"
                                className="form-control"
                                placeholder="Correo"
                                name="email"
                                value={ email }
                                onChange={ handleInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contraseña" 
                                name="password1"
                                value={ password1 }
                                onChange={ handleInputChange }
                            />
                        </div>

                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Repita la contraseña" 
                                name="password2"
                                value={ password2 }
                                onChange={ handleInputChange }
                            />
                        </div>

                        <div className="form-group btn-auth">
                            <input 
                                type="submit" 
                                className="btnSubmit" 
                                value="Crear cuenta" />
                        </div>

                        <Link 
                            // to="/auth/login"
                            to="/login"
                            className="btnSubmit btn-auth"
                        >
                            ¿Ya tienes cuenta?
                        </Link>

                    </form>
                </div>
            </div>
        </div>
    );
};
