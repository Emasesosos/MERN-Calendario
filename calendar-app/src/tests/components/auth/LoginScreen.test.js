import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { MemoryRouter, Route } from 'react-router-dom';
import { LoginScreen } from '../../../components/auth/LoginScreen';
import { startLogin } from '../../../redux/actions/auth';

jest.mock('./../../../redux/actions/auth', () => {
    return {
        startLogin: jest.fn()
    }
})

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={ store }>
        <MemoryRouter initialEntries={['/auth/login']}>
            <LoginScreen />
        </MemoryRouter>
    </Provider>
);

describe('Pruebas en <LoginScreen />', () => {

    test('Debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot(); 

    });

    test('Debe de llamar el dispatch del login', () => {
        
        wrapper.find('input[name="email"]').simulate('change', {
            target: {
                name: 'email',
                value: 'emasesosos2@msn.com'
            }
        });

        wrapper.find('input[name="password"]').simulate('change', {
            target: {
                name: 'password',
                value: '123456'
            }
        });

        wrapper.find('form').at(0).prop('onSubmit')({
            preventDefault(){}
        });

        expect(startLogin).toHaveBeenCalledWith('emasesosos2@msn.com', '123456');

    });
    

});