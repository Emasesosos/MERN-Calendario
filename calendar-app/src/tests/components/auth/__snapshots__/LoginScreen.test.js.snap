// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Debe de mostrarse correctamente 1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1"
    >
      <h3>
        Ingreso
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            placeholder="Correo"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group btn-auth"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Login"
          />
        </div>
        <a
          className="btnSubmit btn-auth"
          href="/auth/register"
          onClick={[Function]}
        >
          Crear Nueva Cuenta
        </a>
        <a
          className="btnSubmit btn-auth mt-3"
          href="/auth/restablish"
          onClick={[Function]}
        >
          ¿Olvidaste tu Contraseña?
        </a>
      </form>
    </div>
  </div>
</div>
`;
